### ISSUE TITLE
[BUG] SfOverlay where content less than 100% height

### ISSUE THREAD

**Issue by infinityredux** [vuestorefront/storefront-ui#1506]
**Describe the bug**
`SfOverlay` will stop at the bottom of the page content, potentially leaving an uncovered bar at the bottom of the screen.

**How to reproduce**
Simplest way to reproduce this that I can think of:
1. Have a VSF theme that uses the component.
2. Ensure that the content of the page is less than the full height.
3. Trigger the overlay.

I might suggest Caybara for step 1 and editing the Home page to remove things for step 2, but that's not a perfect example since Capybara isn't on the latest version of the UI.

**Expected behavior**
`SfOverlay` should cover the entire page.

**Actual behavior**
Obvious bar underneath the footer with the original page background colour.

**Devices (please complete the following information):**
 - Device: Desktop
 - OS: Windows
 - Browser: Opera, Chrome, Firefox, Edge

Doesn't appear to occur on mobile view, but that seems more to do with differences in how menu and search work in that view.

**Workaround**
Simple enough, just add to the style wherever the component is used:
```
::v-deep .sf-overlay {
  bottom: initial;
  height: 100vh;
}
```

**Comment by justyna-13** [comment]
I couldn't reproduce this bug. I had a Home page with only one small component and SfOverlay works fine. Can you send more details? And screenshot how it looks like

**Comment by infinityredux** [comment]
Not sure how well this will show on a white background, because the page background that is showing through is also white, but example of the white bar at the bottom after the end of the footer:
![viewport height before](https://user-images.githubusercontent.com/10075126/97611506-2f475380-1a0e-11eb-96c9-af1811198251.PNG)

Using the workaround above, the image becomes something like this:
![viewport height after](https://user-images.githubusercontent.com/10075126/97611632-556cf380-1a0e-11eb-8c33-83ebd81d2e6e.PNG)

If you're having trouble replicating this with a basic storefront, it might be something about the page structure used by Capybara specifically? (I'm using a custom theme built on top of Capybara.)

**Comment by AdamPawlinski** [comment]
@justyna-13 I reproduced this bug on capybara theme.
![image](https://user-images.githubusercontent.com/32042425/99238709-5bc6e200-27fa-11eb-9b11-d8fc00bb73a0.png) 
And I think the problem is in capybara styles where the overlay has "position: absolute: what makes it dependent on viewport element (with relative position) which in these case has less height than 100%. If we change the overlay style to "position: relative" as it is originally, it behaves as it should.  
![image](https://user-images.githubusercontent.com/32042425/99239391-47cfb000-27fb-11eb-9d0c-89d920d69a42.png)

**Comment by dt2patel** [comment]
We fixed this in Capybara setting the min height of the content to be 100vh (a little more complex math than that but that's the jist). The real reason we did it was more of a UX point of view to always have the footer sit at the bottom of the page if there wasn't enough content to scroll, but I guess it solves this too.

Maybe this is a SFUI bug, but I personally feel like the in-theme solution is more elegant since it doesn't rely on SFUI styling to cover 100% use cases but rather extends it.
